packages/gatsby/src/utils/source-nodes.ts

```diff
@@ -74,20 +35,52 @@ function getStaleNodes(
       )
     }
 
+    if (state.touchNodeOptOutTypes.has(rootNode.internal.type)) {
+      return false
+    }
+
     return !state.nodesTouched.has(rootNode.id)
   })
 }
 
 /**
- * Find all stale nodes and delete them
+ * Find all stale nodes and delete them unless the node type has been opted out of stale node garbage collection.
  */
-function deleteStaleNodes(
-  state: IGatsbyState,
-  nodes: GatsbyIterable<IGatsbyNode>
-): void {
-  const staleNodes = getStaleNodes(state, nodes)
+async function deleteStaleNodes(): Promise<void> {
+  const state = store.getState()
+
+  let deleteCount = 0
+
+  const cleanupStaleNodesActivity =
+    report.createProgress(`clean up stale nodes`)
+  cleanupStaleNodesActivity.start()
```

Comment:
I found this progress helpful since on large sites this part of the process hangs for a long time. I could use `report.verbose` if that's preferrable. But for large sites the process can sit for 5+ minutes with no indication anything is happening.

Reply:
Oh actually unit tests are all failing on this not being a function lol. Should I just remove this?

Reply:
probably not mocked correctly